cardano_node_tests/tests/plutus_common.py

```diff
@@ -283,6 +283,170 @@ class PlutusScriptData:
+MINTING_WRITEBITS_V3 = PlutusScriptData(
+    script_file=MINTING_WRITEBITS_PLUTUS_V3,
+    script_type=clusterlib.ScriptTypes.PLUTUS_V3,
+    execution_cost=ExecutionCost(per_time=90646820, per_space=462457, fixed_cost=UNKNOWN_FIXED_COST)
+)
+
+# ----- All succeeding bitwise tests ----- #
+
+succeeding_minting_bitwise_scripts_v3 = [
+    MINTING_ANDBYTESTRING_V3,
+    MINTING_ORBYTESTRING_V3,
+    MINTING_XORBYTESTRING_V3,
+    MINTING_COMPLEMENTBYTESTRING_V3,
+    MINTING_COUNTSETBITS_V3,
+    MINTING_FINDFIRSTSET_V3,
+    MINTING_READBIT_V3,
+    MINTING_REPLICATEBYTE_V3,
+    MINTING_ROTATEBYTESTRING_V3,
+    MINTING_SHIFTBYTESTRING_V3,
+    MINTING_WRITEBITS_V3
+]
+
+
+# ----- All failing bitwise tests ----- #
+
+FAILING_BITWISE_SCRIPTS_V3 = [
+    "failingReadBitPolicyScriptV3_1.plutus",
+    "failingReadBitPolicyScriptV3_2.plutus",
+    "failingReadBitPolicyScriptV3_3.plutus",
+    "failingReadBitPolicyScriptV3_4.plutus",
+    "failingReadBitPolicyScriptV3_5.plutus",
+    "failingReadBitPolicyScriptV3_6.plutus",
+    "failingReadBitPolicyScriptV3_7.plutus",
+    "failingReadBitPolicyScriptV3_8.plutus",
+    "failingReadBitPolicyScriptV3_9.plutus",
+    "failingReadBitPolicyScriptV3_10.plutus",
+    "failingReadBitPolicyScriptV3_11.plutus",
+    "failingReadBitPolicyScriptV3_12.plutus",
+    "failingReadBitPolicyScriptV3_13.plutus",
+    "failingReadBitPolicyScriptV3_14.plutus",
+    "failingReplicateBytePolicyScriptV3_1.plutus",
+    "failingReplicateBytePolicyScriptV3_2.plutus",
+    "failingReplicateBytePolicyScriptV3_3.plutus",
+    "failingReplicateBytePolicyScriptV3_4.plutus",
+    "failingReplicateBytePolicyScriptV3_5.plutus",
+    "failingReplicateBytePolicyScriptV3_6.plutus",
+    "failingWriteBitsPolicyScriptV3_1.plutus",
+    "failingWriteBitsPolicyScriptV3_2.plutus",
+    "failingWriteBitsPolicyScriptV3_3.plutus",
+    "failingWriteBitsPolicyScriptV3_4.plutus",
+    "failingWriteBitsPolicyScriptV3_5.plutus",
+    "failingWriteBitsPolicyScriptV3_6.plutus",
+    "failingWriteBitsPolicyScriptV3_7.plutus",
+    "failingWriteBitsPolicyScriptV3_8.plutus",
+    "failingWriteBitsPolicyScriptV3_9.plutus",
+    "failingWriteBitsPolicyScriptV3_10.plutus",
+    "failingWriteBitsPolicyScriptV3_11.plutus",
+    "failingWriteBitsPolicyScriptV3_12.plutus",
+    "failingWriteBitsPolicyScriptV3_13.plutus",
+    "failingWriteBitsPolicyScriptV3_14.plutus",
+    "failingWriteBitsPolicyScriptV3_15.plutus",
+    "failingWriteBitsPolicyScriptV3_16.plutus",
+    "failingWriteBitsPolicyScriptV3_17.plutus",
+    "failingWriteBitsPolicyScriptV3_18.plutus",
+    "failingWriteBitsPolicyScriptV3_19.plutus"
+]
+
+# We're not currently checking the costs (and it seems to be difficult when the
+# script fails anyway), so the values here don't really matter.
+UNDETERMINED_COST = ExecutionCost(per_time=1_000_000, per_space=100_000, fixed_cost=1234)
+
+def make_failing_scriptdata (name: str):
```

Comment:
Maybe these need to be rewritten to follow the pattern of the earlier tests, but it was easier to do it this way to get to a point where I could re-run all of the tests without writing a large amount of boilerplate.